guides/keras_cv/3d_object_detection_with_keras_cv.py

```diff
@@ -0,0 +1,367 @@
+`waymo_open_dataset.load` function to load the Waymo Open Dataset into our
+data format.
+"""
+
+from keras_cv.datasets.waymo import load
+from keras_cv.datasets.waymo import build_tensors_for_augmentation
+
+# TODO(ianstenbit): Because we can't distribute WOD, we'll need to load wod_one_frame.tfrecord here
+data_shard = './training-data'
+dataset = load(data_shard)
+
+# By default, WOD point clouds are globally positioned, but for object detection
+# we want them with respect to the vehicle, so we transform them to the vehicle
+# frame of reference.
+dataset = dataset.map(transformer.transform_to_vehicle_frame)
+
+# Because number of points is dynamic, we pad them to make our inputs batchable.
+dataset = dataset.map(transformer.pad_or_trim_tensors)
+
+# Then we can easily reformat the tensors into KerasCV's data format!
+dataset = dataset.map(build_tensors_for_augmentation)
+
+# We use a small batch size here on CPU. Generally, point clouds can be pretty
+# large, so batch sizes are often smaller than in the 2D object detection world.
+dataset = dataset.batch(1)
+
+"""
+Loading up the Waymo Open Dataset can be a bit tricky, but this makes it pretty
+simple!
+
+One important note: Waymo Open Dataset is distributed as TFRecords representing
+a Waymo Open Dataset `Frame` proto. This cannot be deserialized in to Tensors
+inside of the TensorFlow graph, so this can cause CPU throttling during
+training.
+
+Therefore, KerasCV offers a utility for transforming Waymo Open Dataset frames
+into tf.Example records which can be more efficiently loaded into a TF graph
+for later training. The utility can be found at
+https://github.com/keras-team/keras-cv/blob/master/examples/training/object_detection_3d/waymo/serialize_records.py
+
+Next up, let's augment our data! In partnership with Waymo, KerasCV offers a
+set of state-of-the-art 3D augmentations for LIDAR data and 3D boxes. They
+behave like all Keras preprocessing layers, and they're very easy to set up.
+"""
+
+augmentations = keras.Sequential([
+  keras_cv.layers.GlobalRandomFlip(),
+  keras_cv.layers.GlobalRandomRotation(max_rotation_angle_z=3.14),
+])
+
+dataset = dataset.map(augmentations)
+
+"""
+In just a few lines of code, we've augmented our input data using a few of the
+3D augmentations offered in KerasCV. Check out the whole set of augmentations
+at https://github.com/keras-team/keras-cv/tree/master/keras_cv/layers/preprocessing_3d.
+
+Next, we'll create a `MultiHeadCenterPillar` model to train. These models are
+very configurable, and the configuration can be a bit overwhelming at first.
+So let's start by defining (and explaining!) some of the configuration.
+
+For a more in-depth understanding of how the model works, check out
+https://arxiv.org/pdf/2006.11275.pdf.
+"""
+
+"""
+Our model will group points into voxels in 3D space, and we need to specify
+how large these voxels will be. Here, we define the width, length, and height
```

Comment:
it would be kind of fun to include a minecraft picture to "remind" the general public what a Voxel is.  It took me a moment to remember - and I've worked with voxels before. :)